packages/powersync_attachments_stream/lib/src/storage/io_local_storage.dart

```diff
@@ -0,0 +1,127 @@
+/// Local storage adapter for handling file operations on the device filesystem.
+///
+/// This file implements the [IOLocalStorage] class, which provides methods for
+/// saving, reading, deleting, copying, and managing files and directories using
+/// the Dart IO library. It is used as the local storage backend for attachments
+/// in the PowerSync attachments system.
+///
+/// Features:
+/// - Save files from streams (creates directories and all necessary parents dynamically if they do not exist)
+/// - Read files as streams
+/// - Delete files and their metadata
+/// - Copy files and their metadata
+/// - Create and remove directories (creates all necessary parents dynamically)
+/// - Check file existence
+
+import 'dart:async';
+import 'dart:io';
+import 'dart:typed_data';
+import 'package:path/path.dart' as p;
+import '../abstractions/local_storage.dart';
+
+/// Implements [AbstractLocalStorageAdapter] for device filesystem using Dart IO.
+///
+/// Handles file and directory operations for attachments.
+class IOLocalStorage implements AbstractLocalStorageAdapter {
+  final Directory baseDir;
+
+  IOLocalStorage([Directory? baseDir])
+    : baseDir = baseDir ?? Directory.systemTemp;
+
+  File _fileFor(String filePath) => File(p.join(baseDir.path, filePath));
+  File _metaFileFor(String filePath) =>
+      File(p.join(baseDir.path, '$filePath.meta.json'));
```

Comment:
It looks like we're not using `_metaFileFor` anymore outside of copying and deleting? So it seems like we can remove this entirely?